src/dmd/dtoh.d

```diff
@@ -235,6 +235,25 @@ extern(C++) void genCppHdrFiles(ref Modules ms)
     buf.writestring("#if !defined(_d_real)\n");
     buf.writestring("# define _d_real long double\n");
     buf.writestring("#endif\n");
+    buf.writeByte('\n');
+    buf.writestring("#if !defined(_D_ENUM)\n");
+    buf.writestring("# define _D_ENUM\n");
+    buf.writestring("# define BEGIN_ENUM(name, upper, lower) enum class name {\n");
+    buf.writestring("# define BEGIN_ENUM_NUMERIC(type, name, upper, lower) enum class name : type {\n");
+    buf.writestring("# define BEGIN_ENUM_TYPE(type, name, upper, lower) namespace name {\n");
+    buf.writestring("# define BEGIN_ANON_ENUM() enum {\n");
+    buf.writestring("# define BEGIN_ANON_ENUM_NUMERIC(type) enum : type {\n");
+    buf.writestring("# define END_ENUM(name, upper, lower) };\n");
+    buf.writestring("# define END_ENUM_NUMERIC(type, name, upper, lower) };\n");
+    buf.writestring("# define END_ENUM_TYPE(type, name, upper, lower) };\n");
+    buf.writestring("# define END_ANON_ENUM() };\n");
+    buf.writestring("# define END_ANON_ENUM_NUMERIC(type) };\n");
+    buf.writestring("# define ENUM_KEY_NUMERIC(type, name, value, enumName, upper, lower, abbrev) name = value,\n");
+    buf.writestring("# define ENUM_KEY_TYPE(type, name, value, enumName, upper, lower, abbrev) static type const name = value;\n");
+    buf.writestring("# define ANON_ENUM_KEY_NUMERIC(type, name, value) name = value,\n");
+    buf.writestring("# define ENUM_CONSTANT_NUMERIC(type, name, value) enum : type { name = value };\n");
+    buf.writestring("# define ENUM_CONSTANT(type, name, value) static type const name = value;\n");
+    buf.writestring("#endif\n");
```

Comment:
You can write this as simply:
```
buf.writestring("
#if !defined(_D_ENUM)
# define _D_ENUM
...
");
```
in one go. It's a lot easier to read. D is this way because I never liked the C way of not being nice about multiline string literals.

Reply:
I tried many permutations of this; it pollutes the output with extra newlines, improper indentations, and the newlines in the output are dependent on which OS was used to compile DMD (`"\n"` vs `"\r\n"`).